pandas/tests/arithmetic/test_timedelta64.py

```diff
@@ -2089,18 +2148,104 @@ def test_td64arr_pow_invalid(self, scalar_td, box_with_array):
             td1**scalar_td
 
 
-def test_add_timestamp_to_timedelta():
-    # GH: 35897
-    timestamp = Timestamp("2021-01-01")
-    result = timestamp + timedelta_range("0s", "1s", periods=31)
-    expected = DatetimeIndex(
+class TestAddSub:
+    """
+    Addition/subtraction between a timedelta64-valued
+    ExtensionArrays/Indexes/Series/DataFrames, and a timedelta64 scalar or
+    timedelta64-valued ExtensionArray/Index/Series/DataFrame.
+    """
+
+    def test_add_raises_if_result_would_overflow(
+        self,
+        max_td64: TD64_TYPE,
+        positive_td64: TD64_BOX_TYPE,
+    ):
+        with td64_overflow_error():
+            max_td64 + positive_td64
+
+        with td64_overflow_error():
+            positive_td64 + max_td64
+
+    @pytest.mark.parametrize(
+        ["rval", "expected_exs"],
         [
-            timestamp
-            + (
-                pd.to_timedelta("0.033333333s") * i
-                + pd.to_timedelta("0.000000001s") * divmod(i, 3)[0]
-            )
-            for i in range(31)
-        ]
+            (Timedelta(1), does_not_raise()),
+            (Timedelta(2), td64_overflow_error()),
+            (Timedelta.max, td64_overflow_error()),
+        ],
     )
-    tm.assert_index_equal(result, expected)
+    def test_sub_raises_if_result_would_overflow(
```

Comment:
FWIW, I tried rewriting this to keep the test function body constant, and mapping each input value to an appropriate `xfail` marker:

```python
xfail_does_not_raise = pytest.mark.xfail(reason="doesn't raise", strict=True)
xfail_raises_overflow_error = pytest.mark.xfail(
    reason="raises wrong error",
    raises=OverflowError,
    strict=True,
)
# ...
    @pytest.mark.parametrize(
        "rval",
        [
            pytest.param(Timedelta(1), marks=xfail_does_not_raise),
            pytest.param(Timedelta(2), marks=xfail_raises_overflow_error),
            pytest.param(Timedelta.max, marks=xfail_raises_overflow_error),
        ],
    )    
    def test_sub_raises_td64_specific_error_if_result_would_overflow(
        self,
        max_td64: TD64_BOX_TYPE,
        rval: Timedelta,
        td64_type: Type[TD64_TYPE],
    ):
        rvalue = tm.wrap_value(rval, td64_type)
        min_td64 = -1 * max_td64

        with pytest.raises(OutOfBoundsTimedelta, match="too small"):
            min_td64 - rvalue
```

Unfortunately that slowed execution by a factor of ~40 (locally). Decorating the test with a single, catch-all `xfail` marker led to the same results. Passing `run=False` did help, but it was still ~5x slower.